include/lsst/daf/base/DateTime.h

```diff
@@ -59,42 +59,203 @@ class DateTime {
+
+    /**
+     * Get date as a double in a specified representation, such as MJD
+     *
+     * @return the date in the required system, for the requested scale
+     * @param[in] system  desired time system (JD, MJD, or EPOCH)
+     * @param[in] scale  desired time scale (TAI, TT or UTC)
+     * @throw lsst.pex.exceptions.DomainError if scale is UTC and the UTC date is before 1961-01-01
+     */    
     double get(DateSystem system = MJD, Timescale scale = TAI) const;
+
+    /** Get date as an ISO8601-formatted string.
+     *
+     * The returned format is:
+     *    yyyy-mm-ddThh:mm:ss.sssssssssZ
+     * where the final Z is only present if scale is UTC
+     *
+     * @param[in] scale  Desired time scale (TAI, TT or UTC).
+     * @throw lsst.pex.exceptions.DomainError if scale is UTC and the UTC date is before 1961-01-01
+     */
     std::string toString(Timescale scale) const;
 
-    struct tm gmtime(Timescale time) const;
-    struct timespec timespec(Timescale time) const;
-    struct timeval timeval(Timescale time) const;
+    /** Get date as a tm struct, with truncated fractional seconds
+     *
+     * @param[in] scale  desired time scale (TAI, TT or UTC)
+     * @return date as a tm struct
+     * @throw lsst.pex.exceptions.DomainError if scale is UTC and the UTC date is before 1961-01-01
+     */
+    struct tm gmtime(Timescale scale) const;
+
+    /**
+     * Get date as a timespec struct, with time in seconds and nanoseconds
+     *
+     * @param[in] scale  Desired time scale (TAI, TT or UTC)
+     * @return date as a timespec struct
+     * @throw lsst.pex.exceptions.DomainError if scale is UTC and the UTC date is before 1961-01-01
+     */
+    struct timespec timespec(Timescale scale) const;
+
+    /**
+     * Get date as a timeval struct, with time in seconds and microseconds
+     *
+     * @param[in] scale  desired time scale (TAI, TT or UTC)
+     * @return date as a timeval struct
+     * @throw lsst.pex.exceptions.DomainError if scale is UTC and the UTC date is before 1961-01-01
+     */
+    struct timeval timeval(Timescale scale) const;
 
     bool operator==(DateTime const& rhs) const;
 
+    /** Return current time as a DateTime
+     *
+     * Assumes the system clock keeps UTC, as almost all computers do.
+     */
     static DateTime now(void);
 
+    /**
+     * Initialize the leap second table from USNO
+     *
+     * The data can be found here: http://maia.usno.navy.mil/ser7/tai-utc.dat
+     * and is saved in DateTime.cc as static constant `leapString`.
+     *
+     * @param leapString  leap second table from USNO as a single multiline string.
+     */
     static void initializeLeapSeconds(std::string const& leapString);
 
 private:
-    long long _nsecs;  ///< Nanoseconds since Unix epoch
+    long long _nsecs;  ///< TAI nanoseconds since Unix epoch
 
+    /**
+     * Get date as Modified Julian Days in the specified time scale
+     *
+     * @param[in] scale  time scale (TAI, TT or UTC)
```

Comment:
MJD UTC is quite common, e.g. in FITS headers, though I don't recall the exact definition at the moment. In any case I did not add the feature, just moved the documentation.